qadence/backends/pyqtorch/convert_ops.py

```diff
@@ -46,6 +68,94 @@
 
 Tdagger is currently not supported.
 """
+FINITE_DIFF_EPS = 1e-06
+
+
+def finitediff_sampling(
+    f: Callable, x: Tensor, eps: float = FINITE_DIFF_EPS, num_samples: int = 10
+) -> Tensor:
+    def _finitediff(val: Tensor) -> Tensor:
+        return (f(x + val) - f(x - val)) / (2 * val)  # type: ignore
+
+    with no_grad():
+        return mean(cat([_finitediff(val) for val in rand(1) for _ in range(num_samples)]))
+
+
+def finitediff(f: Callable, x: Tensor, eps: float = FINITE_DIFF_EPS) -> Tensor:
+    return (f(x + eps) - f(x - eps)) / (2 * eps)  # type: ignore
```

Comment:
could we move these into a place like `qadence.utils` or something like that? Then it becomes a bit more re-usable for our downstream libraries

Reply:
they moved to qadence/backends/utils.py, for now the scalar version is actually fine. i am planning to move to a more elaborate version in a future MR

Reply:
resolving since we can expand this in a subsequent MR